Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc

```diff
@@ -13,179 +13,85 @@
-	_shape = ((ev.logicalPart()).solid()).shape();
-	_name = ((ev.logicalPart()).ddname()).name();
-	_parents = ev.geoHistory();
-	_volume   = ((ev.logicalPart()).solid()).volume();
-	_density  = ((ev.logicalPart()).material()).density();
-	_weight   = _density * ( _volume / 1000.); // volume mm3->cm3
-	_copy     = ev.copyno();
-	_material = ((ev.logicalPart()).material()).name().fullname();
-}
-
-//----------------------------------------------------------------------------------------------------
-
-DetGeomDesc::DetGeomDesc(DDExpandedView* fv, GeometricEnumType type) : _type(type)
-{
-	_ddd = fv->navPos();
-	_params = ((fv->logicalPart()).solid()).parameters();  
-	_trans = fv->translation();
-	_rot = fv->rotation();
-	_shape = ((fv->logicalPart()).solid()).shape();
-	_name = ((fv->logicalPart()).ddname()).name();
-	_parents = fv->geoHistory();
-	_volume   = ((fv->logicalPart()).solid()).volume();  
-	_density  = ((fv->logicalPart()).material()).density();
-	_weight   = _density * ( _volume / 1000.); // volume mm3->cm3
-	_copy     = fv->copyno();
-	_material = ((fv->logicalPart()).material()).name().fullname();
-}
+DetGeomDesc::DetGeomDesc( DDFilteredView* fv ) :
+  m_trans( fv->translation()),
+  m_rot( fv->rotation()),
+  m_name((( fv->logicalPart()).ddname()).name()),
+  m_params((( fv->logicalPart()).solid()).parameters()),
+  m_copy( fv->copyno()),
+  m_z( fv->geoHistory().back().absTranslation().z())
+{}
 
 //----------------------------------------------------------------------------------------------------
-
-DetGeomDesc::DetGeomDesc(DDFilteredView* fv, GeometricEnumType type) : _type(type)
-{
-	_ddd = fv->navPos();
-	_params = ((fv->logicalPart()).solid()).parameters();
-	_trans = fv->translation();
-	_rot = fv->rotation();
-	_shape = ((fv->logicalPart()).solid()).shape();
-	_name = ((fv->logicalPart()).ddname()).name();
-	_parents = fv->geoHistory();
-	_volume   = ((fv->logicalPart()).solid()).volume();
-	_density  = ((fv->logicalPart()).material()).density();
-	_weight   = _density * ( _volume / 1000.); // volume mm3->cm3
-	_copy     = fv->copyno();
-	_material = ((fv->logicalPart()).material()).name().fullname();
-}
-
-//----------------------------------------------------------------------------------------------------
-
 DetGeomDesc::DetGeomDesc(const DetGeomDesc &ref)
 {
 	(*this) = ref;
 }
 
 //----------------------------------------------------------------------------------------------------
 
-DetGeomDesc& DetGeomDesc::operator= (const DetGeomDesc &ref)
+DetGeomDesc& DetGeomDesc::operator= ( const DetGeomDesc &ref )
 {
-	_ddd			= ref._ddd;
-	_params			= ref._params;
-	_trans			= ref._trans;
-	_rot			= ref._rot;
-	_shape			= ref._shape;
-	_name			= ref._name;
-	_parents		= ref._parents;
-	_volume			= ref._volume;
-	_density		= ref._density;
-	_weight			= ref._weight;
-	_copy			= ref._copy;
-	_material		= ref._material;
-	_geographicalID	= ref._geographicalID;
-	_type			= ref._type;
-
-	return (*this);
+  m_params = ref.m_params;
```

Comment:
A comment here noting that `m_container` is not being assigned would be helpful.